exercises/custom-set/custom-set.spec.js

```diff
@@ -1,115 +1,257 @@
 import CustomSet from './custom-set';
 
 describe('CustomSet', () => {
-  test('can delete elements', () => {
-    const expected = new CustomSet([1, 3]);
-    const actual = new CustomSet([3, 2, 1]).delete(2);
-    expect(actual.eql(expected)).toBe(true);
+  test('size: knows its size', () => {
```

Comment:
`size`, `toList` and `clear` are custom functions which this track has added and not there in the [canonical.json](https://github.com/exercism/problem-specifications/blob/master/exercises/custom-set/canonical-data.json).
Let us try to match the canonical data in terms of sequence and keep these custom ones at the bottom so we get nice progression from easier to more difficult tasks.

Reply:
My thinking here was that we might want to have them build `size` first because they will use it in, at the very least, empty but potentially also in `eql`.

Reply:
RIght. Makes sense 👍

Reply:
I think you two have hit upon an important concern: do we want to force a certain implementation on the user?  Or should we only care about the  public API of the class?  I am more inclined to favor the latter.  

My recollection is that there has been an attempt in Exercism to get away from testing private or helper methods, so I'm leabing toward @tejasbubane 's original request that these tests be removed.  I think the methods can remain in the example, since that's an implementation-specific choice.